src/app/Http/Controllers/CrudFeatures/SaveActions.php

```diff
@@ -12,28 +12,40 @@
     public function getSaveAction()
     {
         $saveAction = session('save_action', config('backpack.crud.default_save_action', 'save_and_back'));
-        $saveOptions = [];
-        $saveCurrent = [
-            'value' => $saveAction,
-            'label' => $this->getSaveActionButtonName($saveAction),
+
+        // Generate list of actions to main action order.
+        $saveOptions = [
+            'save_and_back' => $this->getSaveActionButtonName('save_and_back'),
+            'save_and_edit' => $this->getSaveActionButtonName('save_and_edit'),
+            'save_and_new' => $this->getSaveActionButtonName('save_and_new'),
         ];
 
-        switch ($saveAction) {
-            case 'save_and_edit':
-                $saveOptions['save_and_back'] = $this->getSaveActionButtonName('save_and_back');
-                $saveOptions['save_and_new'] = $this->getSaveActionButtonName('save_and_new');
-                break;
-            case 'save_and_new':
-                $saveOptions['save_and_back'] = $this->getSaveActionButtonName('save_and_back');
-                $saveOptions['save_and_edit'] = $this->getSaveActionButtonName('save_and_edit');
-                break;
-            case 'save_and_back':
-            default:
-                $saveOptions['save_and_edit'] = $this->getSaveActionButtonName('save_and_edit');
-                $saveOptions['save_and_new'] = $this->getSaveActionButtonName('save_and_new');
-                break;
+        // Correctly remove "Save and back" action.
+        if (! $this->crud->hasAccess('list')) {
+            unset($saveOptions['save_and_back']);
+        }
+
+        // Correctly remove "Save and new" action.
+        if (! $this->crud->hasAccess('create')) {
```

Comment:
Isn't there a default save action?

Reply:
Settings have 'save_and_back' as default, but that can't be used if you don't have `list` access, the same is valid for `create` access.